e2e/e2e_tests.py

```diff
@@ -39,105 +42,127 @@ def _stop_containers():
+        self._login()
+        self._create_new_order()
+        self._go_to_order()
+        links = self._wait_for_order_to_complete_and_fetch_resulting_links()
+        self.assertEqual(len(links), 8)
+        download_links = self._get_download_links_from_links(links)
+        for download_link in download_links:
+            self._download_and_test_zip_contents(download_link)
+
+    # Helper methods
+    def _login(self):
+        self.browser.get(self._make_link('/login/'))
+        login_form = self.browser.find_element_by_id('osmaxx-login-form')
+        username = self.browser.find_element_by_id('id_username')
+        password = self.browser.find_element_by_id("id_password")
+        username.send_keys(ADMIN_USER_FOR_TESTS)
+        password.send_keys(ADMIN_PASSWORD_FOR_TESTS)
+        login_form.find_element_by_xpath("//input[@type='submit']").click()
+
+    def _logout(self):
+        self.browser.get(self._make_link('/logout/'))
+
+    def _create_new_order(self):
+        self.browser.get(self._make_link('/orders/new/#new-excerpt'))
+        # wait for page to load
+        WebDriverWait(self.browser, 20).until(
+            expected_conditions.presence_of_element_located((By.ID, "new_excerpt_name"))
+        )
+        for html_id, send_value in self.excerpt_data['form_text_inputs'].items():
+            element = self.browser.find_element_by_id(html_id)
+            element.clear()
+            element.send_keys(send_value)
+        for html_id in self.excerpt_data['checkboxes_to_be_ticked']:
+            self.browser.find_element_by_id(html_id).click()
+        self.browser.find_element_by_xpath("//input[@type='submit']").click()
+
+    def _go_to_order(self):
+        self.browser.get(self._make_link('/orders/'))
+        xpath_link_to_order = '/html/body/div/div/div[2]/div/h3/a'
+        WebDriverWait(self.browser, 60).until(
+            expected_conditions.presence_of_element_located((By.XPATH, xpath_link_to_order))
+        )
+        self.browser.find_element_by_xpath(xpath_link_to_order).click()
+
+    def _wait_for_order_to_complete_and_fetch_resulting_links(self):
+        poll_frequency_in_seconds = 10
+        waited_time_in_seconds = 0
+        while not self._download_finished():
+            waited_time_in_seconds += poll_frequency_in_seconds
+            if waited_time_in_seconds >= self.wait_time_seconds:
+                raise selenium_exceptions.TimeoutException
+            sleep(poll_frequency_in_seconds)
+        links = self._get_download_a_tags()
+        return links
+
+    def _get_download_links_from_links(self, links):
+        download_links_index_start = 0
+        download_links_index_skip = 2
+        number_of_downloads = 4
+        return [links[i].get_attribute('href') for i in range(
+            start=download_links_index_start,
+            stop=number_of_downloads * download_links_index_skip - 1,
+            step=download_links_index_skip,
+        )]
```

Comment:
I noticed we could use slicing here, instead of indexing with a fancy `range`:

    return [link.get_attribute('href') for link in links[download_links_index_start:number_of_downloads * download_links_index_skip - 1:download_links_index_skip]
or even just

    return [link.get_attribute('href') for link in links[download_links_index_start::download_links_index_skip]